zarr/hierarchy.py

```diff
@@ -414,6 +416,78 @@ def arrays(self):
                                  chunk_store=self._chunk_store,
                                  synchronizer=self._synchronizer)
 
+    def visit(self, func):
+        """Run callable on each object name.
+
+        Examples
+        --------
+        >>> from __future__ import print_function, unicode_literals
+        >>> import zarr
+        >>> g1 = zarr.group()
+        >>> g2 = g1.create_group('foo')
+        >>> g3 = g1.create_group('bar')
+        >>> d1 = g1.create_dataset('baz', shape=100, chunks=10, fill_value=0)
+        >>> d2 = g1.create_dataset('quux', shape=200, chunks=20, fill_value=0)
```

Comment:
We can do that. Was thinking about doing something similar. Though the tests explore this as well.

May also drop datasets (or at least disable compression) as the compression levels seem to vary, which would make the doctest annoying to keep.